scripts/gulpfiles/release_notes.js

```diff
@@ -0,0 +1,201 @@
+    Authorization: "bearer " + githubToken,
+  };
+
+
+  var note = `{Insert greeting},
+
+The {Insert Month and Year} release is out! In this release:
+
+{List major features in both core and samples}
+
+🎉 **Kudos** 🎉
+{Thank forums contributors and issue filers}
+
+**Third Party Plugins**
+{List third party plugins that have been released in the last quarter}
+
+**Core contributions**
+{Reorganize into thank-yous and check for missed people}
+${await getContributors('blockly', 'develop', goalDate, headers)}
+
+**Samples contributions**
+{Reorganize into thank-yous and check for missed people}
+${await getContributors('blockly-samples', 'master', goalDate, headers)}
+
+🔨 **Breaking Changes in core** 🔨 
+${await getBreakingChanges(headers)}
+
+Full release notes are below. As always, we welcome bug reports and pull requests!
+
+Cheers,
+{Your Name}
+
+**Issues Closed**:
+${await getClosedIssues(headers)}
+
+**Pull Requests**:
+${await getMergedPrs(headers)}`;
+
+  console.log(note);
+
+  done();
+}
+
+async function getContributors(repo, baseRef, goalDate, headers) {
+  const collaborators = await getCollaborators(repo, headers);
+
+  let cursor = '';
+  const contributors = new Map();
+  do {
+    try {
+      const response = await fetch(baseUrl, {
+        method: "POST",
+        headers: headers,
+        body: makeContributorQuery(repo, baseRef, cursor),
+      });
+      const json = await response.json();
+      const pullRequestData = json.data.repository.pullRequests;
+      const pulls = pullRequestData.nodes;
+      cursor = pullRequestData.pageInfo.endCursor;
+    
+      for (let i = 0; i < pulls.length; i++) {
+        const pull = pulls[i];
+        const createdDate = Date.parse(pull.createdAt);
+        if (createdDate < goalDate) {
+          cursor = '';
+          break;
+        }
+
+        if (collaborators.has(pull.author.login)) {
+          continue;
+        }
+        let contributions = contributors.get(pull.author.login);
+        if (!contributions) {
+          contributions = [];
+          contributors.set(pull.author.login, contributions);
+        }
+        contributions.push({
+          title: pull.title,
+          url: pull.url,
+          number: pull.number
+        })
+      }
+
+    } catch (error) {
+        console.log(error);
+    }
+  } while (cursor);
+
+  let notes = '';
+  for (let [name, contributions] of contributors) {
+    notes += `* ${name}\n`;
+    for (let contribution of contributions) {
+      let {title, url, number} = contribution;
+      notes += `  * ([#${number}](${url})) ${title}\n`;
+    }
+  }
+  return notes;
+}
+
+async function getCollaborators(repo, headers) {
+  try {
+    const response = await fetch(baseUrl, {
+      method: "POST",
+      headers: headers,
+      body: makeCollaboratorQuery(repo),
+    });
+    const json = await response.json();
+    const collaborators =  json.data.repository.collaborators.nodes
+        .map((node) => node.login);
```

Comment:
[Prefer to break lines at a higher syntactic level](https://google.github.io/styleguide/jsguide.html#formatting-where-to-break) (and N.B. parens not required for single-argument arrow functions [but see tip](https://google.github.io/styleguide/jsguide.html#features-functions-arrow-functions) about why you might want them):

```JS
    const collaborators =
        json.data.repository.collaborators.nodes.map(node => node.login);
```
(Better double-check my line lengths, though!)